examples/default/variables.tf

```diff
@@ -22,6 +28,72 @@ variable "resource_group" {
   default     = null
 }
 
+variable "security_group_rules" {
+  description = "A list of security group rules to be added to the default vpc security group"
+  type = list(
+    object({
+      add_ibm_cloud_internal_rules = optional(bool)
+      name                         = string
+      direction                    = string
+      remote                       = string
+      tcp = optional(
+        object({
+          port_max = optional(number)
+          port_min = optional(number)
+        })
+      )
+      udp = optional(
+        object({
+          port_max = optional(number)
+          port_min = optional(number)
+        })
+      )
+      icmp = optional(
+        object({
+          type = optional(number)
+          code = optional(number)
+        })
+      )
+    })
+  )
+  default = [{
+    add_ibm_cloud_internal_rules = true
+    name                         = "allow-all-inbound"
+    direction                    = "inbound"
+    remote                       = "0.0.0.0/0"
+  }]
+
+  validation {
```

Comment:
The validation rules here should be at the root module level. They are currently missing in the variables.tf root module. Once they are in the root module, there is no really strong case to keep them in each example, so could be removed.